### PR TITLE
Adding in handling of fit numbers when fits removed

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
**Description of work.**

When you delete fitted data from the ADS is was causing uncaught exceptions and hard crashes as the deletion was not handled correctly. This updates the number of expected fits as data is removed to prevent this issue. The plotting is also updated so that it checks if plots have been manually removed from graphs.

**To test:**
  * Open up Muon Analysis, load some data and do a couple of fits.
  * Delete the fitted data from the ADS and check that the deletion is handled gracefully. Check that the undo fit button is also handled gracefully.
  * Plot some data and fits, try manually removing some lines and then doing some more fits, check that this is handled gracefully.

*There is no associated issue.*

*This does not require release notes* because this introduces an issue introduced in this release.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/test/Muon/load_file_widget/loadfile_model_test.py [hunk 1]
```diff
@@ -6,14 +6,15 @@
...
+from mantidqt.utils.qt.testing import GuiTest
 
 import six
 
 from Muon.GUI.Common.load_file_widget.model import BrowseFileWidgetModel
 from Muon.GUI.Common.test_helpers.context_setup import setup_context_for_tests
 
 
-class LoadFileWidgetModelTest(unittest.TestCase):
+class LoadFileWidgetModelTest(GuiTest):
```
Comment (Pasarus) [inline scripts/test/Muon/load_file_widget/loadfile_model_test.py]: why do we need to use GUITest here?

**Review by Pasarus — CHANGES_REQUESTED** [review]
Just a small query about some test changes.

File: scripts/test/Muon/load_file_widget/loadfile_model_test.py (hunk 1 above)
Comment (Mantid-Matthew) [reply scripts/test/Muon/load_file_widget/loadfile_model_test.py]: The notify calls emitted when the model is changed go via an invoke method to ensure that any resulting updates to the GUI happen on the GUI thread. This invoke method requires the QApplication to exist.

**Comment by Pasarus** [comment]
@Mantid-Matthew The Python3 failure looks like it is probably legitimate, could you look into that? other than that this looks good :+1:

**Review by Pasarus — APPROVED** [review]

**Pull request merged**